rollup/core/classes/RollupFullRecalcProcessor.cls

```diff
@@ -109,18 +120,19 @@ global abstract without sharing class RollupFullRecalcProcessor extends RollupAs
     }
     this.hasProcessedParentRecords = true;
     Map<Id, SObject> relatedParentRecordsMap = new Map<Id, SObject>(relatedParentRecords);
-    for (SObject parentRecordToReset : this.parentRecordsToClear.values()) {
-      SObject relatedParentRecord = relatedParentRecordsMap.containsKey(parentRecordToReset.Id)
-        ? relatedParentRecordsMap.get(parentRecordToReset.Id)
-        : RollupCurrencyInfo.createNewRecord(parentRecordToReset);
-
-      for (Rollup__mdt meta : this.rollupMetas) {
-        if (relatedParentRecord.getSObjectType().getDescribe().getName() == meta.LookupObject__c) {
-          relatedParentRecord.put(meta.RollupFieldOnLookupObject__c, null);
+    for (Rollup__mdt meta : this.rollupMetas) {
+      Object defaultVal = this.getDefaultValue(meta);
+      for (SObject parentRecordToReset : this.parentRecordsToClear.values()) {
+        if (parentRecordToReset.getSObjectType().getDescribe().getName() != meta.LookupObject__c) {
```

Comment:
I'm not sure if there's any performance difference, but I've started using `toString()` (instead of `getDescribe().getName()`) to _slightly_ shorten up the code in these situations:

```suggestion
        if (parentRecordToReset.getSObjectType().toString() != meta.LookupObject__c) {
```